File: documentation-examples-scala/src/scala/dev/typr/foundations/docs/mariadb/NetworkTypes.scala
package dev.typr.foundations.docs.mariadb

import dev.typr.foundations.{MariaType, MariaTypes}
import dev.typr.foundations.data.maria.{Inet4, Inet6}

@SuppressWarnings(Array("unused"))
object NetworkTypes:
  //start
  val inet4Type: MariaType[Inet4] = MariaTypes.inet4
  val inet6Type: MariaType[Inet6] = MariaTypes.inet6

  val ipv4: Inet4 = Inet4.parse("192.168.1.1")
  val ipv6: Inet6 = Inet6.parse("::1")
  //stop

File: documentation-examples-java/src/java/dev/typr/foundations/docs/mariadb/NetworkTypes.java
package dev.typr.foundations.docs.mariadb;

import dev.typr.foundations.MariaType;
import dev.typr.foundations.MariaTypes;
import dev.typr.foundations.data.maria.Inet4;
import dev.typr.foundations.data.maria.Inet6;

@SuppressWarnings("unused")
public class NetworkTypes {
  // start
  MariaType<Inet4> inet4Type = MariaTypes.inet4;
  MariaType<Inet6> inet6Type = MariaTypes.inet6;

  Inet4 ipv4 = Inet4.parse("192.168.1.1");
  Inet6 ipv6 = Inet6.parse("::1");
  // stop
}
